xcm/xcm-executor/src/traits/should_execute.rs

```diff
@@ -46,21 +67,59 @@ impl ShouldExecute for Tuple {
 		instructions: &mut [Instruction<Call>],
 		max_weight: Weight,
 		weight_credit: &mut Weight,
-	) -> Result<(), ()> {
+	) -> Result<(), RejectReason> {
+		let mut reason = RejectReason::UnexpectedMessageFormat;
 		for_tuples!( #(
-			match Tuple::should_execute(origin, instructions, max_weight, weight_credit) {
+			reason = match Tuple::should_execute(origin, instructions, max_weight, weight_credit) {
 				Ok(()) => return Ok(()),
-				_ => (),
-			}
+				Err(e) => e,
+			};
 		)* );
 		log::trace!(
 			target: "xcm::should_execute",
-			"did not pass barrier: origin: {:?}, instructions: {:?}, max_weight: {:?}, weight_credit: {:?}",
+			"did not pass barrier: origin: {:?}, instructions: {:?}, max_weight: {:?}, weight_credit: {:?}, reason: {:?}",
 			origin,
 			instructions,
 			max_weight,
 			weight_credit,
+			reason,
 		);
-		Err(())
+		Err(reason)
+	}
+}
+
+/// Trait to determine whether the execution engine is suspended from executing a given XCM.
+///
+/// The trait method is given the same parameters as `ShouldExecute::should_execute`, so that the
+/// implementer will have all the context necessary to determine whether or not to suspend the
+/// XCM executor.
+///
+/// Can be chained together in tuples to have multiple rounds of checks. If all of the tuple
+/// elements returns false, then execution is not suspended. Otherwise, execution is suspended
+/// if any of the tuple elements returns true.
+pub trait CheckSuspension {
+	fn is_suspended<Call>(
```

Comment:
Maybe what should happen here is to put the `is_suspended` method inside of `ShouldExecute`, that way we don't need to have a new trait just for this method.